controllers/clusterenvironment_controller.go

```diff
@@ -423,6 +423,60 @@ func (r *ClusterEnvironmentReconciler) finalizeClusterEnvironment(ctx context.Co
 	return nr.ReconcileNamespaces(ctx)
 }
 
+func (r *ClusterEnvironmentReconciler) CreateServiceCatalog(ctx context.Context, ce *primazaiov1alpha1.ClusterEnvironment) error {
+	l := log.FromContext(ctx)
+	l.Info("Service catalog not found in cluster environment")
+
+	var rsl primazaiov1alpha1.RegisteredServiceList
+	lo := client.ListOptions{Namespace: ce.Namespace}
+	if err := r.Client.List(ctx, &rsl, &lo); err != nil {
+		return err
+	}
+	var scs []primazaiov1alpha1.ServiceCatalogService
+	for _, rs := range rsl.Items {
+		var addServiceToCatalog bool
+		if rs.Status.State == primazaiov1alpha1.RegisteredServiceStateAvailable {
+			if rs.Spec.Constraints != nil {
+				if envtag.Match(ce.Spec.EnvironmentName, rs.Spec.Constraints.Environments) {
+					addServiceToCatalog = true
+				}
+			} else {
+				addServiceToCatalog = true
+			}
+		}
+		if addServiceToCatalog {
```

Comment:
this may be simplified

```suggestion
		if rs.Status.State == primazaiov1alpha1.RegisteredServiceStateAvailable &&
		    (rs.Spec.Constraints == nil || envtag.Match(ce.Spec.EnvironmentName, rs.Spec.Constraints.Environments)) {
```

Reply:
Yeah that is better